src/main/java/uk/ac/sanger/sccp/stan/request/LibraryPrepRequest.java

```diff
@@ -0,0 +1,119 @@
+package uk.ac.sanger.sccp.stan.request;
+
+import uk.ac.sanger.sccp.stan.request.ReagentTransferRequest.ReagentTransfer;
+import uk.ac.sanger.sccp.stan.request.SlotCopyRequest.SlotCopyDestination;
+import uk.ac.sanger.sccp.stan.request.SlotCopyRequest.SlotCopySource;
+import uk.ac.sanger.sccp.utils.BasicUtils;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A request to record transfer, dual index and amplification ops.
+ * @author dr6
+ */
+public class LibraryPrepRequest {
+    private String workNumber;
+    private List<SlotCopySource> sources;
+    private SlotCopyDestination destination;
+    private List<ReagentTransfer> reagentTransfers;
+    private String reagentPlateType;
+    private List<SlotMeasurementRequest> slotMeasurements;
+
+    /**
+     * The work number to associate with these operations.
+     */
+    public String getWorkNumber() {
+        return this.workNumber;
+    }
+
+    public void setWorkNumber(String workNumber) {
+        this.workNumber = workNumber;
+    }
+
+    /**
+     * The source labware for this request.
+     */
+    public List<SlotCopySource> getSources() {
+        return this.sources;
+    }
+
+    public void setSources(List<SlotCopySource> sources) {
+        this.sources = sources;
```

Comment:
Are we validating sources to have a length of 2 in the 'Library Amplification and Generation' context?

Reply:
I don't understand this question. What context do you mean and why should we validate to have two sources?

Reply:
I got it clarified from previous comment. Thank you.